drivers/FileHandle.h

```diff
@@ -100,43 +124,20 @@ class FileHandle {
      *    0 on success or un-needed,
      *   -1 on error
      */
-    virtual int fsync() = 0;
-
-    virtual off_t flen() {
-        lock();
-        /* remember our current position */
-        off_t pos = lseek(0, SEEK_CUR);
-        if(pos == -1) {
-            unlock();
-            return -1;
-        }
-        /* seek to the end to get the file length */
-        off_t res = lseek(0, SEEK_END);
-        /* return to our old position */
-        lseek(pos, SEEK_SET);
-        unlock();
-        return res;
-    }
-
-    virtual ~FileHandle() {};
-
-protected:
-
-    /** Acquire exclusive access to this object.
-     */
-    virtual void lock() {
-        // Stub
-    }
+    MBED_DEPRECATED_SINCE("mbed-os-5.4", "Replaced by FileLike::sync")
+    virtual int fsync() { return sync(); }
```

Comment:
Oh, that was unintentional, are those lock/unlock functions used at all? It seems like files should probably not be shared accross threads, and if you did you would want to lock at a higher level anyways?

Could they be added back in if we needed them in a backwards compatible manner?